orphanet/disease_parser.py

```diff
@@ -0,0 +1,59 @@
+import lxml.etree as ET
+from dataclasses import dataclass, asdict, field
+import h5py
+import json
+import numpy as np
+from typing import List, Generator
+
+
+@dataclass
+class Disease:
+    disorder_id: int
+    disorder_type: str
+    orpha_code: int
+    name: str
+    expert_link: str
+    meta_data: dict = field(default_factory=dict)
+
+
+class OrphanetDiseaseXMLParser:
+    def __init__(
+        self, disease_xml_file: str, out_dir: str
+    ):  # tree structure of the xml file
+        self.xml_file = disease_xml_file
+        self.out_dir = out_dir
+        self.tree = ET.parse(self.xml_file)
+        self.root = self.tree.getroot()
+        self.meta_data = self.extract_meta_data()
+
+    def extract_meta_data(self) -> List[str]:
+        metadata = []
+        elements_of_interest = ["Licence", "ClassificationList"]
```

Comment:
the given elements_of_interest are wrongly strcutured! The metadata inline is mostly in between <Availability></Availability>. You have to create an extra for loop for ClassificationList to get Classification id. 

`  <Availability> 
    <Licence>
      <FullName lang="en">Creative Commons Attribution 4.0 International</FullName>
      <ShortIdentifier>CC-BY-4.0</ShortIdentifier>
      <LegalCode>https://creativecommons.org/licenses/by/4.0/legalcode</LegalCode>
    </Licence>
  </Availability>`

Also for other xml files, we have to create dynamic and elastic label detection. We cannot define each tag and subcategory for each new xml file.